김민준/섹션4-완전탐색/01_자릿수의-합.js

```diff
@@ -0,0 +1,37 @@
+function solution(N, NUMBER_LIST) {
+  let eachDigitSumObj = {};
+
+  // 각 자릿수의 합을 value로 객체에 초기화
+  NUMBER_LIST.forEach((num) => {
+    eachDigitSumObj[num] = num
+      .toString()
+      .split('')
+      .map(Number)
+      .reduce((acc, cur) => acc + cur);
+  });
+
+  // 각 자릿수의 합중 최댓값 구하기
+  let maxSum = 0;
+  maxSum = Math.max(...Object.values(eachDigitSumObj));
+  
+  // 각 자릿수의 합이 최댓값과 같은 수 뽑아내기
+  let maxSumList = [];
+  maxSumList = Object.entries(eachDigitSumObj).filter(entry => entry[1] === maxSum);
```

Comment:
```suggestion
  maxSumList = Object.entries(eachDigitSumObj).filter([key, value]=> value === maxSum);
```
filter 콜백에서 인자값 구조분해 할당도 가능합니당 :)

Reply:
> 

오 미처 생각못했네요! 확실히 저렇게 할 때 좀 맘에 안들었는데 감사합니당!